platform-sdk/docs/proposals/TSS-Library/TSS-Library.md

```diff
@@ -0,0 +1,807 @@
+|--------------------|--------------------------------------------------------|
+| Designers          | Austin, Cody, Edward, Rohit, Maxi, <br/> Platform team | 
+| Functional Impacts | Platform team. Release engineering. DevOps             |
+| Related Proposals  | TSS-Roster, TSS-Ledger-Id                              |
+
+
+## Purpose and Context
+A threshold signature scheme (TSS) aims to enable a threshold number of participants (shareholders) to securely and efficiently generate succinct aggregate signatures.
+In our TSS implementation, a static public key should be produced that doesn't change even when the number of participants in the scheme varies,
+and it can be used to verify the aggregate of a set of partial signatures produced by a threshold number of private shares.
+
+This is important for producing proofs that are easily consumable and verifiable by external entities.
+
+This proposal covers the implementation of all necessary components to provide the consensus node and block node with
+the functionality to sign and verify blocks using a Threshold Signature Scheme (TSS) and EC Cryptography.
+
+The related proposal, TSS-Ledger-Id, provides an overview of the process and background for TSS and how it impacts the platform’s functionality.
+
+This proposal assumes no relation with the platform and defines a generic components that any consumer can integrate.
+It only assumes that there exists a channel to connect participants, where the identity of the message sender has been previously validated.
+
+The process of sending messages through that channel and receiving the responses is outside the scope of this proposal.
+Additionally, participants will need access to each other's public key. While the generation of the public/private keys is included in this proposal,
+the distribution aspect, the loading, and the in-memory interpretation from each participant are outside the scope of this proposal.
+
+### Glossary
+- **TSS (Threshold Signature Scheme)**: A cryptographic signing scheme in which a minimum number of parties (reconstruction threshold) must collaborate
+  to produce an aggregate signature that can be used to sign messages and an aggregate public key that can be used to verify that signature.
+- **Groth 21**: Publicly verifiable secret sharing and resharing schemes that enable secure and efficient distribution and management of secret shares,
+  with many possible use cases supporting applications in distributed key generation and threshold signatures.
+  Uses Shamir's secret sharing, ElGamal, and Zero-Knowledge Proofs.
+- **Distribute key generation**: Aims to solve the problem of getting n parties able to cryptographically sign and verify signatures in the presence of some corruption threshold in a decentralized network. 
+To do so, this algorithm generates a public key, and a secret key of which no single party knows, but has some share of.
```

Comment:
Can we change this to be something like:

"To do so, this algorithm generates a public key and a private key. The private key is distributed as key-splits such that no single party knows the private key."